File: IEEE/Xtreme 17.0/Wargames.cpp
#include <bits/stdc++.h>
#define int long long
#define all(a) a.begin(), a.end()
#define vii vector<pair<int, int>>
#define fastio                   \
    ios::sync_with_stdio(false); \
    cin.tie(0);                  \
    cout.tie(0)
using namespace std;

template <typename T>
void print(vector<T> &array) {
    for (auto &it : array) {
        cout << it << " ";
    }
    cout << "\n";
}

vector<string> card_values = {"2", "3", "4", "5", "6", "7", "8", "9", "T", "J", "Q", "K", "A"};

signed main() {
    fastio;
    int n;
    cin >> n;
    cin.ignore();
    string s;
    string token;

    map<string, int> cards;
    for (int i = 0; i < card_values.size(); i++) {
        cards[card_values[i]] = i + 1;
    }
    for (int i = 0; i < n; i++) {
        getline(cin, s);
        stringstream ss1(s);
        queue<string> player1;
        while (getline(ss1, token, ' ')) {
            player1.push(token);
        }
        getline(cin, s);
        stringstream ss2(s);
        queue<string> player2;
        while (getline(ss2, token, ' ')) {
            player2.push(token);
        }
        int iterations = 0;
        while (!player1.empty() && !player2.empty() && iterations < 2700) {
            string a = player1.front();
            string b = player2.front();

            player1.pop();
            player2.pop();

            if (cards[a] > cards[b]) {
                player1.push(b);
            } else if (cards[a] < cards[b]) {
                player2.push(a);
            } else {
                player1.push(a);
                player2.push(b);
            }
            iterations++;
        }
        if (player1.empty()) {
            cout << "player 2"
                 << "\n";
        } else if (player2.empty()) {
            cout << "player 1"
                 << "\n";
        } else {
            cout << "draw"
                 << "\n";
        }
        // print(player1);
        // print(player2);
    }

    return 0;
}

File: IEEE/Xtreme 17.0/wargames.py
from queue import Queue

card_values = {
    "2": 1,
    "3": 2,
    "4": 3,
    "5": 4,
    "6": 5,
    "7": 6,
    "8": 7,
    "9": 8,
    "T": 9,
    "J": 10,
    "Q": 11,
    "K": 12,
    "A": 13,
}

n = int(input())

for i in range(n):
    player1 = Queue(input().split())
    player2 = Queue(input().split())
    p1_cards = {
        "2": 0,
        "3": 0,
        "4": 0,
        "5": 0,
        "6": 0,
        "7": 0,
        "8": 0,
        "9": 0,
        "T": 0,
        "J": 0,
        "Q": 0,
        "K": 0,
        "A": 0,
    }
    p2_cards = {
        "2": 0,
        "3": 0,
        "4": 0,
        "5": 0,
        "6": 0,
        "7": 0,
        "8": 0,
        "9": 0,
        "T": 0,
        "J": 0,
        "Q": 0,
        "K": 0,
        "A": 0,
    }

    for i in player1:
        p1_cards[i] += 1
    for i in player2:
        p2_cards[i] += 1

    draw = False
    while len(player1) > 0 and len(player2) > 0:
        a = player1[0]
        b = player2[0]
        if p1_cards[a] == len(player1) and p2_cards[a] == len(player2):
            draw = True
            break

        player1.popleft()
        p1_cards[a] -= 1
        player2.popleft()
        p2_cards[b] -= 1
        if card_values[a] > card_values[b]:
            player1.append(b)
            p1_cards[b] += 1
        elif card_values[a] < card_values[b]:
            player2.append(a)
            p2_cards[a] += 1
        else:
            player1.append(a)
            player2.append(b)
            p1_cards[a] += 1
            p2_cards[b] += 1

    if draw == True:
        print("draw")
    elif len(player1) > 0:
        print("player 1")
    else:
        print("player 2")
